src/index.ts

```diff
@@ -131,7 +133,7 @@ export async function retry<T> (
       options.beforeAttempt(context, options);
     }
 
-    if (context.aborted) {
+    if (context.aborted || (options.signal && options.signal.aborted)) {
```

Comment:
Could it be simplified like this?

```suggestion
    if (context.aborted || options.signal?.aborted) {
```

Reply:
Babel is not configured to understand `?.` or `??`. So I used fallback myself.